modules/mediagoBidAdapter.js

```diff
@@ -0,0 +1,458 @@
+        cat: [], // todo
+        publisher: {
+          // todo
+          id: domain,
+          name: domain,
+        },
+      },
+      imp: items,
+      tmax: timeout,
+    };
+    return c;
+  } else {
+    return null;
+  }
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  // aliases: ['ex'], // short code
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    // console.log('mediago', {
+    //   bid
+    // });
+    if (bid.params.token) {
+      globals['token'] = bid.params.token;
+    }
+    return !!bid.params.token;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {Array}  validBidRequests an an array of bids
+   * @param {Object} bidderRequest  The master bidRequest object
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    let payload = getParam(validBidRequests, bidderRequest);
+
+    const payloadString = JSON.stringify(payload);
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL + globals['token'],
+      data: payloadString,
+    };
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bids = getProperty(serverResponse, 'body', 'seatbid', 0, 'bid');
+    const cur = getProperty(serverResponse, 'body', 'cur');
+
+    const bidResponses = [];
+    for (let bid of bids) {
+      let impid = getProperty(bid, 'impid');
+      if (itemMaps[impid]) {
+        let bidId = getProperty(itemMaps[impid], 'req', 'bidId');
+        const bidResponse = {
+          requestId: bidId,
+          cpm: getProperty(bid, 'price'),
+          width: getProperty(bid, 'w'),
+          height: getProperty(bid, 'h'),
+          creativeId: getProperty(bid, 'crid'),
+          dealId: '',
+          currency: cur,
+          netRevenue: true,
+          ttl: TIME_TO_LIVE,
+          // referrer: REFERER,
+          ad: getProperty(bid, 'adm'),
+          //   adserverTargeting: {
+          //     granularityMultiplier: 0.1,
+          //     priceGranularity: "pbHg",
+          //     pbMg: "0.01",
+          //   },
+          //   pbMg: "0.01",
+          //   granularityMultiplier: 0.1,
+          //   priceGranularity: "pbHg",
+        };
+        bidResponses.push(bidResponse);
+      }
+    }
+
+    return bidResponses;
+  },
+
+  /**
+   * Register bidder specific code, which will execute if bidder timed out after an auction
+   * @param {data} Containing timeout specific data
+   */
+  onTimeout: function (data) {
```

Comment:
I'm also unsure if onTimeout, onBidWon and onSetTargeting are necessary if you do not have anything happen within them. Did you have plans for these?

Reply:
got it